CONTRIBUTING.md

```diff
@@ -111,73 +143,80 @@ Make sure you document each commit and squash them appropriately. You can use th
 * Scala's documentation on [Git Hygiene](https://github.com/scala/scala/tree/v2.12.0-M3#git-hygiene)
 * Play's documentation on [Working with Git](https://www.playframework.com/documentation/2.4.4/WorkingWithGit#Squashing-commits)
 
-Documentation
--------------
-
-Documentation fixes and contributions are as much welcome as to the source code itself. Visit [the website project](https://github.com/sbt/website) to learn about how to contribute.
-
 Build from source
 =================
 
-1. Install the current stable binary release of sbt (see [Setup]), which will be used to build sbt from source.
-2. Get the source code.
-
-		$ git clone git://github.com/sbt/sbt.git
-		$ cd sbt
+### Branch to work against
 
-3. The default branch is the development branch [1.0.x](https://github.com/sbt/sbt/tree/1.0.x), which contains the latest code for the next major sbt release.  To build a specific release or commit, switch to the associated tag.  The tag for the latest stable release is [v0.13.13](https://github.com/sbt/sbt/tree/v0.13.13):
+sbt uses two branches for development:
 
-		$ git checkout v0.13.13
+- Development branch: `1.x` (this is also called "master")
+- Stable branch: `1.$MINOR.x`, where `$MINOR` is current minor version (e.g. `1.1.x` during 1.1.x series)
 
-	Note that sbt is always built with the previous stable release.  For example, the [1.0.x](https://github.com/sbt/sbt/tree/1.0.x) branch is built with 0.13.13 and the [v0.13.13](https://github.com/sbt/sbt/tree/v0.13.13) tag is built with 0.13.12.
+If you're working on a bug fix, it's a good idea to start with the `1.$MINOR.x` branch, since we can always safely merge from stable to `1.x`, but not other way around.
 
-4. To build the launcher and publish all components locally,
+### Instruction to build all modules from source
 
-		$ sbt
-		> publishLocal
+1. Install the current stable binary release of sbt (see [Setup]), which will be used to build sbt from source.
+2. Get the source code.
 
-5. To use this locally built version of sbt, copy your stable `~/bin/sbt` script to `~/bin/xsbt` and change it to use the launcher jar at `<sbt>/launch/target/sbt-launch.jar`.
+   ```
+   $ mkdir sbt-modules
+   $ cd sbt-modules
+   $ for i in sbt io util librarymanagement zinc; do \
+     git clone https://github.com/sbt/$i.git && (cd $i; git checkout -b 1.1.x origin/1.1.x)
+   done
+   $ cd sbt
+   $ ./sbt-allsources.sh
+   ```
 
-	Directory `target` is removed by `clean` command. Second solution is using the artifact stored in the local ivy repository.
+3. To build and publish all components locally,
```

Comment:
Test changes to modules using `sbt test` in the module directory, and/or by following other the instructions in the module's README.

Once you're ready to integration test changes to module(s) with SBT:

```
cd sbt 
./sbt-allsources.sh
 sbt:sbtRoot> publishLocalAllModule
```